core/src/main/java/com/novoda/espresso/FontScaleTestRule.java

```diff
@@ -0,0 +1,75 @@
+package com.novoda.espresso;
+
+import android.os.SystemClock;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+class FontScaleTestRule implements TestRule {
+
+    private final FontScaleSetting fontScaleSetting;
+    private final FontScale fontScale;
+
+    FontScaleTestRule(FontScaleSetting fontScaleSetting, FontScale fontScale) {
+        this.fontScaleSetting = fontScaleSetting;
+        this.fontScale = fontScale;
+    }
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new FontScaleStatement(base, fontScaleSetting, fontScale);
+    }
+
+    private static class FontScaleStatement extends Statement {
+
+        private static final int SLEEP_TO_WAIT_FOR_SETTING_MILLIS = 100;
+        private static final int MAX_RETRIES_TO_WAIT_FOR_SETTING = 15;
+
+        private final Statement baseStatement;
+        private final FontScaleSetting scaleSetting;
+        private final FontScale scale;
+
+        FontScaleStatement(Statement baseStatement, FontScaleSetting scaleSetting, FontScale scale) {
+            this.baseStatement = baseStatement;
+            this.scaleSetting = scaleSetting;
+            this.scale = scale;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
```

Comment:
I prefer to use [TestWatcher](https://junit.org/junit4/javadoc/4.12/org/junit/rules/TestWatcher.html) if my rule is simple enough. It also has lifecycle methods and also handles possible exceptions for you nicely.

Reply:
I _just_ read about test watcher and was going to use it for the screenshot rule :D I can change it then, since this isn't going to affect the public api